index.js

```diff
@@ -1,604 +1,676 @@
-  var currentIndex = array.length, temporaryValue, randomIndex ;
-
-  // While there remain elements to shuffle...
-  while (0 !== currentIndex) {
-
-    // Pick a remaining element...
-    randomIndex = Math.floor(Math.random() * currentIndex);
-    currentIndex -= 1;
-
-    // And swap it with the current element.
-    temporaryValue = array[currentIndex];
-    array[currentIndex] = array[randomIndex];
-    array[randomIndex] = temporaryValue;
-  }
-
-  return array;
-}
-
-function findPlayerIndexByRole(role) {
-	var hits = [];
-	for(var i = 0; i < players.length; i++) {
-		if(players[i].playerInitialRole == role) {
-			hits.push(i);
-		}
-	}
-	return hits;
-}
-
-function getAllPlayerNames() {
-	var names = [];
-	for(var i = 0; i < players.length; i++) {
-		names.push(players[i].playerUserName);
-	}
-	return names;
-}
-
-//returns -1 if cant find
-function findPlayerIndexByName(name) {
-	for(var i = 0; i < players.length; i++) {
-		if(players[i].playerUserName == name) {
-			return i;
-		}
-	}
-	return -1;
-}
+var app = require('express')();
+var http = require('http').Server(app);
+var io = require('socket.io')(http);
+var animal = require('animal-id');
+app.get('/', function(req, res){
+  res.sendfile('index.html');
+});
+
+
+var newPlayerId = 0;
+var players = [];
+var gameState = new gameState();
+var gamePhases = {
+	SETUP : 'setup',
+	NIGHT : 'night',
+	DAY : 'day',
+	ENDED : 'ended' 
+}
+
+var gamePhase = gamePhases.SETUP;
+var roles = {
+    WEREWOLF : 'werewolf',
+	MASON : 'mason',
+    SEER : 'seer',
+	ROBBER : 'robber',
+	TROUBLEMAKER : 'troublemaker',
+	DRUNK : 'drunk',
+	INSOMNIAC : 'insomniac',
+	VILLAGER : 'villager',
+	TANNER : 'tanner',
+	HUNTER : 'hunter'
+}
+
+var playerActions = {
+	DONOTHING : 'donothing',
+	VIEW : 'view',
+	SWAP : 'swap',
+	VIEWANDSWAP : 'viewandswap'
+}
+
+
+io.on('connection', function(socket){
+  var socketId = socket.id;
+
+  players.push(new playerState(socketId));
+  
+  //push the initial set of buttons
+  
+  //io.sockets.connected[socketId].emit('update gamebuttons', 'test');
+	
+  socket.on('chat message', function(msg){
+    processChatMessage(msg, socketId);
+  });
+  
+  socket.on('disconnect', function(){
+	  var removedPlayer = removePlayer(socketId);
+	  console.log('removed player', removedPlayer); 
+  });  
+  
+});
+
+//todo i know this is bad practice. Ask a javascript person how to do this better
+var setupHTML = "";
+
+http.listen(3000, function(){
+  console.log('listening on *:3000');
+});
+
+
+//define player variables
+function playerState(socket) {
+	//sets color to a random color
+	this.playerNameColor = '#' + Math.random().toString(16).substring(2, 8);
+	this.playerSocket = socket;
+	this.playerNumber = newPlayerId;
+	newPlayerId += 1;
+	this.playerUserName = animal.getId();
+	this.playerInitialRole = roles.VILLAGER;
+	this.playerCurrentRole = roles.VILLAGER;
+	this.playerAction;
+	this.playerTokens = [];
+	this.playerCurrentVote = players.length - 1;

```

Comment:
Later on I check if something < 0, I add players.length to it. Now that I think about it I think starting day phase we should set playerCurrentVote instead of at the very beginning, but this should work if people don't leave.